### PR TITLE
HDDS-5556. GrpcReplication Client might fail in SCM HA Cluster

### PR DESCRIPTION
**Pull request by vivekratnavel**
## What changes were proposed in this pull request?

Add all CA Certs to trust manager while creating GRPC Replication Server between datanodes.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5556

## How was this patch tested?

Tested with docker-compose by bringing up the following containers initially in ozonesecure-ha environment:

```
docker-compose up -d kdc kms scm1.org scm2.org scm3.org recon om1 om2 om3 datanode1
// To change the SCM leader node
docker-compose stop scm1.org

// Start datanode2 to get its cert signed by new SCM leader (scm2.org)
docker-compose up -d datanode2

// Change leader again
docker-compose start scm1.org
docker-compose stop scm2.org

// Start dn3
docker-compose up -d datanode3

// put a file
docker-compose exec om bash
kinit -kt /etc/security/keytabs/om.keytab om/om
ozone sh vol create vol1
ozone sh bucket create vol1/bucket1
ozone sh key put /vol1/bucket1/key1 <path_to_file>

// stop dn3 and remove
docker-compose stop datanode3
docker-compose rm datanode3

// start dn3 as a fresh container
docker-compose up -d datanode3
```

After this, verified that SCM scheduled replication of the container to datanode3 and it succeeded without any errors.

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java [hunk 1]
```diff
@@ -51,6 +52,8 @@
 
   private SecurityConfig secConf;
 
+  private ConfigurationSource config;
```
Comment (bharatviswa504) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationServer.java:55]: Unused one

**Review by bharatviswa504 — COMMENTED** [review]
Overall LGTM. One minor comment

**Comment by vivekratnavel** [comment]
@bharatviswa504 Thanks for the review. PTAL.

**Review by bharatviswa504 — APPROVED** [review]

**Comment by bharatviswa504** [comment]
Thank You @vivekratnavel for the contribution.

**Pull request merged**